src/event_listener.rs

```diff
@@ -137,6 +146,60 @@ pub enum FlushReason {
     CatchUpAfterErrorRecovery = 14,
 }
 
+impl Drop for FlushJobInfo<'_> {
+    fn drop(&mut self) {
+        unsafe {
+            ffi::rocksdb_table_properties_destroy(self.table_properties.as_ptr());
+        }
+    }
+}
+
+impl FlushJobInfo<'_> {
+    pub fn get_user_collected_property(&self, key: impl CStrLike) -> Option<&CStr> {
+        unsafe {
+            let key_cstring = key.into_c_string().unwrap();
+            let value_ptr = ffi::rocksdb_table_properties_get_user_collected_property(
+                self.table_properties.as_ptr(),
+                key_cstring.as_ptr(),
+            );
+
+            if value_ptr.is_null() {
+                return None;
+            }
+
+            let value_string = CStr::from_ptr(value_ptr);
+            Some(value_string)
+        }
+    }
+
+    pub fn get_user_collected_property_keys(&self, prefix: Option<impl CStrLike>) -> Vec<&CStr> {
```

Comment:
potentially overkill, an impl CStrLike should be enough and users who don't want prefix can pass `""` akin to what you are doing internally really.